OpenOversight/app/gmail_client.py

```diff
@@ -0,0 +1,117 @@
+import base64
+from email.mime.text import MIMEText
+
+from apiclient import errors
+from flask import current_app, render_template
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+
+from OpenOversight.app.config import BaseConfig
+
+
+class Email:
+    """Base class for all emails."""
+
+    def __init__(self, body: str, subject: str, receiver: str):
+        self.body = body
+        self.receiver = receiver
+        self.subject = subject
+
+    def create_message(self):
+        message = MIMEText(self.body, "html")
+        message["to"] = self.receiver
+        message["from"] = BaseConfig.OO_SERVICE_EMAIL
+        message["subject"] = self.subject
+        return {"raw": base64.urlsafe_b64encode(message.as_bytes()).decode()}
+
+
+class AdministratorApprovalEmail(Email):
+    def __init__(self, receiver: str, **kwargs):
+        subject = (
+            f"{current_app.config.get('OO_MAIL_SUBJECT_PREFIX')} New User Registered"
+        )
+        body = render_template("auth/email/new_registration.html", **kwargs)
+        super().__init__(body, subject, receiver)
+
+
+class ChangeEmailAddressEmail(Email):
+    def __init__(self, receiver: str, **kwargs):
+        subject = (
+            f"{current_app.config.get('OO_MAIL_SUBJECT_PREFIX')} Confirm Your Email "
+            f"Address"
+        )
+        body = render_template("auth/email/change_email.html", **kwargs)
+        super().__init__(body, subject, receiver)
+
+
+class ConfirmAccountEmail(Email):
+    def __init__(self, receiver: str, **kwargs):
+        subject = (
+            f"{current_app.config.get('OO_MAIL_SUBJECT_PREFIX')} Confirm Your Account"
+        )
+        body = render_template("auth/email/confirm.html", **kwargs)
+        super().__init__(body, subject, receiver)
+
+
+class ConfirmedUserEmail(Email):
+    def __init__(self, receiver: str, **kwargs):
+        subject = (
+            f"{current_app.config.get('OO_MAIL_SUBJECT_PREFIX')} New User Confirmed"
+        )
+        body = render_template("auth/email/new_confirmation.html", **kwargs)
+        super().__init__(body, subject, receiver)
+
+
+class ResetPasswordEmail(Email):
+    def __init__(self, receiver: str, **kwargs):
+        subject = (
+            f"{current_app.config.get('OO_MAIL_SUBJECT_PREFIX')} Reset Your Password"
+        )
+        body = render_template("auth/email/reset_password.html", **kwargs)
+        super().__init__(body, subject, receiver)
+
+
+class GmailClient(object):
```

Comment:
This is great! Surprisingly little code required to actually do the actual email sending.

I am fine with keeping things as they are, but I am wondering if it makes sense to split the GmailClient and the Email class into two files, which would make it easier for others to know what to replace if they want to use a different email solution (they would only need to replace the gmail client, not have to touch the email class).

Reply:
That's fair! Let me move them and rename the client.

Reply:
I am going to save moving the emails to their own file until I can address this issue:

If I put the emails in the `models.py` file we end up creating a circular dependency:
```python
ImportError while loading conftest '/usr/src/app/OpenOversight/tests/conftest.py'.
OpenOversight/tests/conftest.py:22: in <module>
    from OpenOversight.app import create_app, models
OpenOversight/app/__init__.py:21: in <module>
    from .email_client import EmailClient
OpenOversight/app/email_client.py:6: in <module>
    from OpenOversight.app.models import Email
OpenOversight/app/models.py:16: in <module>
    from . import login_manager
E   ImportError: cannot import name 'login_manager' from partially initialized module 'OpenOversight.app' (most likely due to a circular import) (/usr/src/app/OpenOversight/app/__init__.py)
```